cmd/router/main.go

```diff
@@ -330,9 +332,11 @@ func compilePatterns(patterns []string) ([]*regexp.Regexp, error) {
 var (
 	jsonGraph              = flag.String("graph-json", "", "serialized json graph def")
 	compiledHeaderPatterns []*regexp.Regexp
+	timeout                int64
 )
 
 func main() {
+	flag.Int64Var(&timeout, "termination-grace-period", 300, "wait timeout period for in-flight requests during server shutdown")
```

Comment:
@yuzisun / @rachitchauhan43 I think we need to introduce a new field for specifying the timeout period. WDYT ?

Reply:
In serverless mode, i am seeing, due to knative, `terminationGracePeriodSeconds: 300` is getting added to router pods.
But yes, I agree, just like we let users control this in ISVCs, we should let users control this in IG too.

Reply:
@yuzisun : We are on KServe 0.10.1 and observing this issue. If this is the fix, I would need your help in back-porting this to 0.10.1 or better if we could release 0.10.1.1 for this hotfix.